core/src/org/javarosa/core/model/condition/EvaluationContext.java

```
@@ -348,11 +349,19 @@ private void expandReferenceAccumulator(TreeReference sourceRef, DataInstance so
                         //Just by getting here we're establishing a position for evaluating the current
                         //context. If we break, we won't push up the next one
                         positionContext[predIndex]++;
+                        boolean passed = false;
 
                         //test the predicate on the treeElement
                         //EvaluationContext evalContext = new EvaluationContext(this, treeRef);
                         EvaluationContext evalContext = rescope(treeRef, positionContext[predIndex]);
-                        Object o = xpe.eval(sourceInstance, evalContext);
+                        Object o;
+                        try {
+                            o = xpe.eval(sourceInstance, evalContext);
+                        } catch (XPathTypeMismatchException e) {
+                            // Yay, the predicate's so jacked it failed to evaluate!
+                            passedAll = false;
```

Should we fail silently here or not?